### PR TITLE
[Dy2Stat]Support Mixed Precision training in @to_static

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features
### PR changes

Others
### Describe

[Dy2Stat]Support Mixed Precision training in @to_static.

Examle:

```python
# suppose the forward has beed decorated by @to_static
model = MNIST()

adam = AdamOptimizer(
            learning_rate=0.001, parameter_list=mnist.parameters())

scaler = paddle.amp.GradScaler(init_loss_scaling=1024)

for batch in batch_iter_reader:
    with paddle.amp.auto_cast():
            prediction, acc, avg_loss = mnist(img, label=label)

    scaled = scaler.scale(avg_loss)
    scaled.backward()
    scaler.minimize(adam, scaled)
    mnist.clear_gradients()
```

> 1. 此 PR 为主体功能PR，后续会添加更多模型单测验证和完善@to_static 支持混合精度训练的功能

### CODE REVIEW

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Comment by zhangbo9674** [comment]
LGTM

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -2035,6 +2035,11 @@ def __init__(self,
                 del op_attrs[role_var_name]
 
             if len(self.desc.type()) != 0:
+                # NOTE(Aurelius84): prog.clone() will lead that var.op is always None,
```
Comment (chenwhql) [inline python/paddle/fluid/framework.py:2038]: why not fix this program in prog.clone?

**Review by chenwhql — APPROVED** [review]
LGTM for framework.py

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/framework.py:2038]: prog.clone() will call `Operator()`

**Pull request merged**